gusto/initialisation_tools.py

```diff
@@ -5,10 +5,70 @@
 
 from __future__ import absolute_import
 from firedrake import MixedFunctionSpace, TrialFunctions, TestFunctions, \
+    TestFunction, TrialFunction, \
     FacetNormal, inner, div, dx, ds_b, ds_t, DirichletBC, \
     Expression, Function, Constant, \
     LinearVariationalProblem, LinearVariationalSolver, \
-    NonlinearVariationalProblem, NonlinearVariationalSolver, split
+    NonlinearVariationalProblem, NonlinearVariationalSolver, split, solve
+
+
+def incompressible_hydrostatic_balance(state, b0, p0, top=False, params=None):
+
+    # get F
+    v = TrialFunction(state.Vv)
+    w = TestFunction(state.Vv)
+
+    unp1 = state.xnp1.split()[0]
+    dim = unp1.ufl_element().value_shape()[0]
+    bc = ("0.0",)*dim
```

Comment:
Can just use `zero(unp1.ufl_shape)`, then no need for `Expression(bc)` in the bcs below.